components/heap/third_party/dartino/cmpctmalloc.c

```diff
@@ -549,11 +556,11 @@ IRAM_ATTR static void *create_allocation_header(
 void cmpct_test_buckets(void)
 {
     size_t rounded;
-    unsigned bucket;
-    // Check for the 8-spaced buckets up to 128.
-    for (unsigned i = 1; i <= 128; i++) {
+    unsigned bucket; // Check for the 8-spaced buckets up to 128.
+    for (unsigned i = 0; i <= 128; i++) {
         // Round up when allocating.
         bucket = size_to_index_allocating(i, &rounded);
+        ASSERT(bucket >= 0 && bucket < NUMBER_OF_BUCKETS);
```

Comment:
Given that 'bucket' is unsigned it would be quite surprising if it wasn't `>= 0`.
Don't necessarily mind the check, but afraid that compilers will warn that the expression is always true.